File: sdks/python/examples/events/event.py
from examples.simple.worker import EVENT_KEY, hatchet, step1
from hatchet_sdk.clients.events import PushEventOptions

SCOPE = "foobar"
hatchet.filters.create(
    workflow_id=step1.id,
    expression="input.should_skip == false",
    scope=SCOPE,
    payload={"test": "test"},
)

hatchet.event.push(
    EVENT_KEY,
    {"should_skip": True},
    PushEventOptions(
        ## If no scope is provided, all events pushed will trigger workflows
        ## (i.e. the filter will not apply)
        scope=SCOPE,
        additional_metadata={
            "shouldSkipThis": True,
        },
    ),
)

hatchet.event.push(
    EVENT_KEY,
    {"should_skip": False},
    PushEventOptions(
        ## If no scope is provided, all events pushed will trigger workflows
        ## (i.e. the filter will not apply)
        scope=SCOPE,
        additional_metadata={
            "shouldSkipThis": False,
        },
    ),
)

File: pkg/examples/run/event.go
package main

import (
	"context"

	"github.com/google/uuid"
	"github.com/joho/godotenv"

	v1_workflows "github.com/hatchet-dev/hatchet/examples/go/workflows"
	"github.com/hatchet-dev/hatchet/pkg/client"
	"github.com/hatchet-dev/hatchet/pkg/client/rest"
	v1 "github.com/hatchet-dev/hatchet/pkg/v1"
)

func event() {
	err := godotenv.Load()
	if err != nil {
		panic(err)
	}

	hatchet, err := v1.NewHatchetClient()

	if err != nil {
		panic(err)
	}
	// > Pushing an Event
	err = hatchet.Events().Push(
		context.Background(),
		"simple-event:create",
		v1_workflows.SimpleInput{
			Message: "Hello, World!",
		},
	)
	// !!

	// > Create a filter
	payload := map[string]interface{}{
		"main_character":       "Anna",
		"supporting_character": "Stiva",
		"location":             "Moscow",
	}

	_, err = hatchet.Filters().Create(
		context.Background(),
		rest.V1CreateFilterRequest{
			WorkflowId: uuid.New(),
			Expression: "input.shouldSkip == false",
			Scope:      "foobarbaz",
			Payload:    &payload,
		},
	)
	// !!

	if err != nil {
		panic(err)
	}

	// > Skip a run
	skipPayload := map[string]interface{}{
		"shouldSkip": true,
	}
	skipScope := "foobarbaz"
	err = hatchet.Events().Push(
		context.Background(),
		"simple-event:create",
		skipPayload,
		client.WithFilterScope(&skipScope),
	)
	// !!

	if err != nil {
		panic(err)
	}

	// > Trigger a run
	triggerPayload := map[string]interface{}{
		"shouldSkip": false,
	}
	triggerScope := "foobarbaz"
	err = hatchet.Events().Push(
		context.Background(),
		"simple-event:create",
		triggerPayload,
		client.WithFilterScope(&triggerScope),
	)
	// !!

	if err != nil {
		panic(err)
	}
}
